### PR TITLE
Bench 338 implement tags endpoint

### PR DESCRIPTION
**Pull request by JoeCSykes**

### CODE REVIEW

File: answerking_app/models/serializers.py [hunk 1]
```diff
@@ -20,22 +21,24 @@
 from answerking_app.utils.mixins.ApiExceptions import ProblemDetails
 
 MAXNUMBERSIZE = 2147483647
+name_regex_str = "^[a-zA-Z !]+$"
+desc_regex_str = "^[a-zA-Z .!,#]+$"
```
Comment (NogaAD) [inline answerking_app/models/serializers.py]: Why are we not allowing digits in the regex?

File: answerking_app/tests/test_unit/test_unit_tag_serializer.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        self.assertEqual(actual, expected)
+
+    def test_tag_serializer_retired_field_content(self):
+        test_tag: Tag = Tag.objects.get(name=self.test_tag_data["name"])
+        test_serializer_data: ReturnDict = TagSerializer(test_tag).data
+        expected: bool = test_tag.retired
+        actual: bool = test_serializer_data["retired"]
+        self.assertEqual(actual, expected)
+
+    def test_tag_serializer_products_field_content(self):
```
Comment (NogaAD) [inline answerking_app/tests/test_unit/test_unit_tag_serializer.py:70]: Maybe add a test for invalid/nonexistent product id.

File: answerking_app/urls/tag_urls.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+from functools import partial
+
+from django.urls import path
+
+from answerking_app.views import tag_views
+
+urlpatterns: list[partial] = [
```
Comment (NogaAD) [inline answerking_app/urls/tag_urls.py:7]: What about tag/{id}/products endpoint? or is that a different ticket?

Comment (JoeCSykes) [reply answerking_app/urls/tag_urls.py:7]: It will be a different ticket as It is currently not on the API contract

Comment (JoeCSykes) [reply answerking_app/urls/tag_urls.py:7]: so a list of product ids associated with that tag is now included in the tag/{id} response so tag/{id}/products isn't needed any more

File: answerking_app/models/serializers.py (hunk 1 above)
Comment (JoeCSykes) [reply answerking_app/models/serializers.py]: added numbers to regex and changed the unit tests for this

File: answerking_app/tests/test_unit/test_unit_tag_serializer.py (hunk 2 above)
Comment (JoeCSykes) [reply answerking_app/tests/test_unit/test_unit_tag_serializer.py:70]: added 2 tests; one where id is invalid, and another for when id is not found

**Review by NogaAD — APPROVED** [review]

**Review by MichaelOSullivanAnswer — APPROVED** [review]

**Pull request merged**